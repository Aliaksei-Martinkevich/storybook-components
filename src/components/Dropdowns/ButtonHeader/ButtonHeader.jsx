import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './ButtonHeader.scss';
import Button, { BUTTON_THEMES } from '../../Buttons/Button/Button.jsx';

export const BUTTON_HEADER_THEMES = BUTTON_THEMES;

const DropdownButton = ({ text, className, theme, expanded }) =>
  (<Button
    text={text}
    theme={theme}
    className={classNames('button_dropdown-header', className, {
      'button_dropdown-header_expanded': expanded,
      'button_dropdown-header_collapsed': !expanded,
    })}
  />);

DropdownButton.defaultProps = {
  expanded: false,
  className: '',
  theme: BUTTON_THEMES.PRIMARY,
};

DropdownButton.propTypes = {
  text: PropTypes.string.isRequired,
  expanded: PropTypes.bool,
  className: PropTypes.string,
  theme: PropTypes.oneOf(Object
      .getOwnPropertyNames(BUTTON_HEADER_THEMES)
      .map(name => BUTTON_HEADER_THEMES[name])),
};

export default DropdownButton;
