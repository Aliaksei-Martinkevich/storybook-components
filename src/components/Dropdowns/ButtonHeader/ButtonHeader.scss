.button_dropdown-header {
    &_collapsed {
        background-image: url('../../../../assets/icons/svg/collapse_arrow-white.svg');
        background-repeat: no-repeat;
        background-position: calc(100% - 20px) center;
        background-size: 15px;
    }

    &_expanded {
        background-image: url('../../../../assets/icons/svg/expand_arrow-white.svg');
        background-repeat: no-repeat;
        background-position: calc(100% - 20px) center;
        background-size: 15px;
    }
}