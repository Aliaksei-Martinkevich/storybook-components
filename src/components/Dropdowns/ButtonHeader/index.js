import React from 'react';
import PropTypes from 'prop-types';
import './index.css';
import Buttons from '../../Buttons';
const PrimaryButton = Buttons.PrimaryButton;

const DropdownButton = ({className = '', text = ''}) => {
  return (<PrimaryButton text={text} className={`dropdown-button-header`}/>);
};

DropdownButton.defaultProps = {
  className: '',
}

DropdownButton.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string,
}

export default DropdownButton;
