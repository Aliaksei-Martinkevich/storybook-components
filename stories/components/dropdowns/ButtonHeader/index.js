import React from 'react';
import { storiesOf } from '@storybook/react';

import Components from '../../../../src/components'
const ButtonHeader = Components.Dropdowns.ButtonHeader;

export default storiesOf('Dropdown Button Header', module)
  .add('Expanded', () => <div className='expanded'><ButtonHeader text='Expanded header' /></div>)
  .add('Collapsed', () => <div className='collapsed'><ButtonHeader text='Collapsed header' /></div>);