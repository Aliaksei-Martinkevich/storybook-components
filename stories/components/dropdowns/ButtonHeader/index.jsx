import React from 'react';
import { storiesOf } from '@storybook/react';

import ButtonHeader from '../../../../src/components/Dropdowns/ButtonHeader/ButtonHeader.jsx';

export default storiesOf('Dropdown Button Header', module)
  .add('Expanded', () => <div className="expanded"><ButtonHeader text="Expanded header" /></div>)
  .add('Collapsed', () => <div className="collapsed"><ButtonHeader text="Collapsed header" /></div>);
