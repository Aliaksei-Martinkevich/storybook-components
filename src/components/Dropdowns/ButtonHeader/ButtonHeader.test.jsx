import React from 'react';
import renderer from 'react-test-renderer';
import ButtonHeader, { BUTTON_HEADER_THEMES } from './ButtonHeader.jsx';

describe('Button header', () => {
  it('Should be collapsed', () => {
    const customClassName = 'class-name-example';
    const text = 'Some text';
    const component = renderer.create(
      <ButtonHeader
        theme={BUTTON_HEADER_THEMES.DEFAULT}
        className={customClassName}
        text={text}
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should be expanded', () => {
    const customClassName = 'class-name-example';
    const text = 'Some text';
    const component = renderer.create(
      <ButtonHeader
        className={customClassName}
        text={text}
        expanded
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
