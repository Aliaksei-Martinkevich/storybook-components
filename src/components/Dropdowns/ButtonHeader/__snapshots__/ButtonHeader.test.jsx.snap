// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button header Should be collapsed 1`] = `
<button
  className="button button_theme-default button_dropdown-header class-name-example collapsed"
  disabled={false}
  onClick={undefined}
  title="Some text"
>
  <div
    className="button-content"
  >
    <span
      className="button-label butten-content__label"
    >
      Some text
    </span>
  </div>
</button>
`;

exports[`Button header Should be expanded 1`] = `
<button
  className="button button_theme-primary button_dropdown-header class-name-example expanded"
  disabled={false}
  onClick={undefined}
  title="Some text"
>
  <div
    className="button-content"
  >
    <span
      className="button-label butten-content__label"
    >
      Some text
    </span>
  </div>
</button>
`;
